presto-docs/src/main/sphinx/admin/resource-groups.rst

```diff
@@ -9,287 +9,412 @@ Except for the limit on queued queries, when a resource group runs out of a reso
+while database-based resource management is more complex to set up but offers more flexibility and dynamic changes.
+
+File Resource Group Manager
+---------------------------
+
+The file resource group manager in PrestoDB is a way to manage resources using a JSON configuration file.
+This file contains definitions of all resource groups and rules for selecting the appropriate resource group
+for a given query.
+
+To set up a file-based resource group manager:
+
+1. Create the file ``etc/resource-groups.properties``.
+
+2. Set the ``resource-groups.configuration-manager`` property to ``file``.
+
+3. Create a JSON file: This file should contain the definitions for your resource groups.
+   Each resource group can specify things like the maximum memory, the maximum queued queries,
+   and the maximum running queries.
+
+4. In the ``etc/resource-groups.properties`` file, add a line that specifies the location of your JSON file.
+   Set the ``resource-groups.config-file`` property to ``<file_path>``, where ``<file_path>`` is the path to
+   your JSON file. The resource-groups.config-file is a configuration file used to define resource groups in Presto.
+   Resource groups are a mechanism for managing resources in Presto. For example of a resource-groups.json file, follow
+   resources-groups-example.json.
+
+.. code-block:: text
+
+    resource-groups.configuration-manager=file
+    resource-groups.config-file=etc/resource-groups.json
+
+5. Restart the Presto server. The new resource groups will take effect after the restart.
+
+Database Resource Group Manager
+-------------------------------
+
+The database resource group manager in PrestoDB is a way to manage resources using a relational database.
+
+To set up a database-based resource group manager:
+
+1. Create the file ``etc/resource-groups.properties``.
+
+2. Set the ``resource-groups.configuration-manager`` property to ``db``.
+
+3. Set up a relational database: The database should be accessible by Presto.
+   It will be used to store the configurations of your resource groups.
+
+4. Create tables for resource groups and selection rules: You need to create tables in the
+   database that will store the definitions of your resource groups and the rules for selecting
+   the appropriate resource group for a given query.
+
+5. To specify the database in the Presto configuration, add a line in etc/resource-groups.properties that
+   specifies the JDBC URL of your database: resource-groups.config-db-url = <jdbc_url>, where <jdbc_url> is the
+   JDBC URL of your database.
+
+Note : The supported databases are MySQL, PostgreSQL, and Oracle.
+
+With the Database Resource Group Manager, changes to the configuration in the database take effect immediately
+and do not require a restart of the Presto server. This allows for more flexibility and
+dynamic changes to the resource group configurations.
+
+etc/resource-groups.properties should be similar to this following example:
```

Comment:
```suggestion
``etc/resource-groups.properties`` should be similar to this following example:
```

Also, I suggest moving this line and the code block example to after the Note and before "With the Database Resource Group Manager, ". This code block will not need to be indented when it is moved because it is not part of a step.